### ISSUE TITLE
Bus error on Solaris 10 when calling getinfo

### ISSUE THREAD

**Issue by GoogleCodeExporter** [maximemaher/pyodbc#86]
```
What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?

PyODBC 2.1.6 on Solaris 10.

Please provide any additional information below.

Lines 503-505 of connection.cpp attempt to read a long into
the address space of an unsigned small int.

SQLUSMALLINT infotype;    
if (!PyArg_ParseTuple(args, "l", &infotype))        
    return 0;

```

Original issue reported on code.google.com by `mike.t.k...@gmail.com` on 19 Jan 2010 at 1:33

Attachments:
- [diff.txt](https://storage.googleapis.com/google-code-attachments/pyodbc/issue-86/comment-0/diff.txt)

**Comment by GoogleCodeExporter** [comment]
```
I believe this is fixed in 2.1.8.  Can you test this?
```

Original comment by `mkleehammer` on 6 Sep 2010 at 5:39
- Changed state: **Investigating**

**Comment by GoogleCodeExporter** [comment]
```
I'm not the reporter of this issue, but I was able to confirm the bug in 2.1.7 
on our solaris 10 install. The 2.1.8 version has fixed the problem with 
getinfo. Thanks!
```

Original comment by `myer0...@gmail.com` on 18 Nov 2010 at 4:28

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 2.1.8.

```

Original comment by `mkleehammer` on 20 Nov 2010 at 11:04
- Changed state: **Verified**

**Comment by GoogleCodeExporter** [comment]
Original comment by `mkleehammer` on 21 Nov 2010 at 4:45
- Changed state: **Complete**
